Supplementary Table 1.  Concepts in sample preparation and library construction that can influence study design, analysis and interpretation.
- Pros and cons of each.  e.g. This strategy may hurt alternative splicing analysis.  If you only care about gene expression you might choose this strategy.

| Strategy/Concept | Comments |
|------------------|----------|
| RNA integrity/degradation | Discuss strategies to avoid to RNA degradation (RNAse inhibitor, chelating agents, RNA storage products (RNAlater), RNA isolation from protein, temperature).  What to consider if your RNA is already degraded (avoid polyA selection, alter fragmentation strategy, consider cDNA capture). |
| PolyA selection vs. total RNA vs. Ribominus (Figure 3a) | PolyA: https://www.biostars.org/p/73163/#73191 <br>  Total RNA: http://core-genomics.blogspot.com/2012/03/what-is-total-rna.html <br> Prior to sequencing, RNA must be isolated. Depending on what the target for sequencing is, there are two main options for isolating RNA: PolyA selection and Total RNA Capture. Each has its advantages.  In PolyA selection, a oligo-dT probe is used to capture the poly-A tail at the 3’ end of mature, processed mRNA. In doing PolyA selection, one is essentially enriching for mature mRNA only, leaving behind the pre-processed mRNA as well as other non-coding RNA.  In a total-RNA capture, the targets for capture are less biased, and a more even distribution of RNA’s is obtained. This procedure will yield sequence reads for non-coding RNA, pre-processed RNA, and other functional RNA molecules like tRNA and rRNA. While this data tends to be noisier, it also gives a more accurate representation of the RNA population within the cell. | 
| Fragmentation | |
| Size selection | |
| Linear amplification | |
| Exome capture of an RNA-seq library | |
| Library normalization | |
| Strand specific versus unstranded RNA-seq libraries (Figure 3b) | |
| Indexing and pooling of multiple RNA-seq libraries | |
| Narrow versus broad size selection versus small-RNA | |



